### ISSUE TITLE
help needed for module RegistryDict.py

### ISSUE THREAD

**Issue by quintijn** [dictation-toolbox/natlink#15]
The Natlink configure program needs to do a few things in the registry. The module RegistryDict.py could handle this, and for the simple case of one new key in one place it still works. When adding more subkeys to one key, the script fails.

In order to be more versatile, I would like this module to be ported to python3 with essential changes:
-change win32api to winreg (this is fairly trivial)
-change subclassing object or dict (which is not clear) to collections.UserDict
-make additions of a dictionary into a key/subkey possible.

The current code is in the natlink core directory, and some efforts are already made (RegistryDictUserDictWinreg.py). 

A start for a unittest program has been made in the PyTest directory of Natlink.

Hope somebody can help here. I get stuck with understanding all the special __ functions.

**Comment by dougransom** [comment]
Assign it to me.

**Comment by quintijn** [comment]
Yeah, I did. Doug, if it does not work well, we will find an easier, but more restricted solution.

**Comment by quintijn** [comment]
Hi Doug, are you giving your efforts a chance? If so, please report (no hurry), otherwise, we can try to do the limited registry operations with simpler more dedicated functions...

**Comment by kb100** [comment]
I think that this might be the wrong abstraction for our problem. A UserDict is fundamentally a wrapper around an ordinary dict, evidenced by the required .data field (which has not been added in your implementation). Perhaps you could massage it into fitting an abc MutableMapping, but I don't think this is right either because things like popitem dont make sense in the registry (what is a reference to a deleted registry entry?). Although the registry uses "keys" and "values", they don't seem to corrrspond much to dictionary keys and values. A registry key is more like two dictionaries, one for data (values) and one for subkeys. Making the registry have dictionary syntax is admirable, but I don't think it really fits the model.

Given that natlink only needs to set/query 5 or 6 keys in the registry total, (as far as i can tell) all of which are strings, perhaps it would make more sense to remove RegistryDict altogether, and simply use winreg instead.
